File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/Tanimlamalar/Drugs/DrugsColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace MuayeneYonetimPortali.Tanimlamalar.Columns;

[ColumnsScript("Tanimlamalar.Drugs")]
[BasedOnRow(typeof(DrugsRow), CheckNames = true)]
public class DrugsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int DrugId { get; set; }
    [EditLink]
    public string Name { get; set; }
    public bool IsActive { get; set; }
    public string Barcode { get; set; }
    public string CompanyName { get; set; }
    public string AtcCode { get; set; }
    public string AtcName { get; set; }
}

File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/ServerTypes/Tanimlamalar/DrugsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DrugsRow } from "./DrugsRow";

export interface DrugsColumns {
    DrugId: Column<DrugsRow>;
    Name: Column<DrugsRow>;
    IsActive: Column<DrugsRow>;
    Barcode: Column<DrugsRow>;
    CompanyName: Column<DrugsRow>;
    AtcCode: Column<DrugsRow>;
    AtcName: Column<DrugsRow>;
}

export class DrugsColumns extends ColumnsBase<DrugsRow> {
    static readonly columnsKey = 'Tanimlamalar.Drugs';
    static readonly Fields = fieldsProxy<DrugsColumns>();
}